conda/gateways/connection/__init__.py

```diff
@@ -2,7 +2,8 @@
 # SPDX-License-Identifier: BSD-3-Clause
 try:
     from requests import ConnectionError, HTTPError, Session
-    from requests.adapters import BaseAdapter, HTTPAdapter
+    from requests.adapters import DEFAULT_POOLBLOCK, BaseAdapter
+    from requests.adapters import HTTPAdapter as BaseHTTPAdapter
```

Comment:
This change of import will likely break some 3rd party conda packages, let's keep using the name `HTTPAdapter` here and then import in the new `adapters/http.py` using a new local name.

```suggestion
    from requests.adapters import DEFAULT_POOLBLOCK, BaseAdapter, HTTPAdapter
```